nixos/modules/services/monitoring/prometheus/exporters/node.nix

```diff
@@ -26,17 +26,29 @@ in
         Collectors to disable which are enabled by default.
       '';
     };
+    textfileDir = mkOption {
+      type = types.path;
+      default = "/var/lib/prometheus-node-exporter/textfile";
+      description = lib.mdDoc ''
+        Path that the textfile collector will read from, if it is not disabled.
+        This directory will _not_ automatically be created.
```

Comment:
Why not create the directory automatically? Are users supposed to `mkdir` manually or configure a `systemd.tmpfiles.rules`?

Reply:
Ok, I see that node-exporter doesn't have a state directory -- I assumed that the base directory (`/var/lib/prometheus-node-exporter`) existed already, but I was wrong.

How does node-exporter behave when given `--collector.textfile.directory ${cfg.textfileDir}"` by default and the directory doesn't exist? Any errors or warnings?

Reply:
> What do you think, should I add a systemd.tmpfiles.rules entry for this?

I'm not sure. I personally would, but maybe someone has a different use case that I don't know about?

(I do think it's a bad idea to pass `--collector.textfile.directory=...` to a missing directory as a default setting though. I'm surprised node_exporter doesn't complain about the directory being missing.)